advocacy_docs/edb-postgres-ai/migration-etl/data-migration-service/getting_started/preparing_db/preparing_postgres_source_databases.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: "Preparing Postgres source databases"
+---
+
+<div id="preparing_postgres_sources" class="registered_link"></div>
+
+Configuring Postgres for EDB Data Migration Service (EDB DMS) requires administrative privileges. Create a change data capture (CDC) migration role with limited privileges for data migration.
+
+Execute SQL statements with psql or a similar client.
+
+To connect to the source database using `psql`:
+
+```shell
+psql -h <PG_HOST> -p <PG_PORT> -U <PG_USERNAME> -d <PG_DB_NAME>
+```
+
+Where:
+ - `<PG_DB_NAME>` is the name of the Postgres database source to connect to.
+ - `<PG_HOST>` is the Postgres database host.
+ - `<PG_PORT>` is the Postgres database port.
+ - `<PG_USERNAME>` is an administrative user who can create and grant roles, alter ownership of tables to migrate, and create a replication slot.
+
+This command prompts you for the password associated with `<PG_USERNAME>`.
+
+## Postgres database configuration
+
+To prepare the source Postgres database configuration:
+1. [Verify Postgres configuration](#verify-postgres-configuration).
+1. [Create new roles and grant acccess for CDC migration](#create-new-roles-and-grant-acccess-for-cdc-migration).
+1. [Grant `SELECT` on source tables to the CDC migration role](#grant-select-on-source-tables-to-the-cdc-migration-role).
+1. [Create a logical replication slot](#create-a-logical-replication-slot)
+
+### Verify Postgres configuration
+
+Verify or set these configuration entries for Postgres.
+
+1. Ensure `wal_level` is configured as `logical`.
+
+    The CDC migration process leverages Postgres logical decoding. Setting `wal_level` to `logical` enables logical decoding of the Postgres write-ahead log (WAL).
+
+2. Ensure `max_wal_senders` is configured appropriately.
+
+    If EDB Data Migration Service migration is the first streaming client for your database, set `max_wal_senders` to at least `1`. Other streaming clients might be present. Consult your DBA for the appropriate value for streaming client connectivity.
+
+3. Ensure `max_replication_slots` is configured appropriately.
+
+    `max_replication_slots` must be at least `1` for the CDC migration process. This value can be higher if your organization uses Postgres replication.
+
+    See the [Postgres replication documentation](https://www.postgresql.org/docs/current/runtime-config-replication.html) for more information.
+
+4. Ensure `max_wal_size` is configured for adequate WAL LSN lifetime.
+
+    Set the `max_wal_size` value large enough that production traffic is generating mostly timed checkpoints and not requested checkpoints based on WAL size.
```

Comment:
@mlewandowski-edb @ksnavely I don't have a strong opinion about this.

Based on this article(https://www.enterprisedb.com/blog/tuning-maxwalsize-postgresql) looks like there is not a unique value for max_wal_size, it depends on each customer and how they are using it.

We add this check as part of this ticket: https://enterprisedb.atlassian.net/browse/ET-333

Reply:
Ok. 
@gvasquezvargas For now, let's not worry about changing anything in the documentation related to `max_wal_size'.

@jperozoa I think the real issue is that the config validation script indicates a failure if the value isn't set to > 8 GB. We should revisit the config validation script. Even if we want to recommend > 8 GB in most production use cases, we may want to flag that as a warning instead of a failure or at least provide some additional info why it failed validation.